spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/source/TestORCDataFrameWrite.java

```diff
@@ -30,4 +34,24 @@ protected void configureTable(Table table) {
         .set(TableProperties.DEFAULT_FILE_FORMAT, FileFormat.ORC.toString())
         .commit();
   }
+
+  @Test
+  @Override
+  public void testUnknownListType() {
```

Comment:
Minor: It doesn't seem worth copying these tests so many times. Maybe have a `supportsUnknownInListAndMap` instead?

Reply:
Unless you feel strongly, I'd rather keep it for now to make sure that we raise the right `Exception`. Also, this allows us to implement List and Map seperately.